src/manifest.rs

```diff
@@ -65,262 +65,153 @@ impl From<bare_version::Error> for ManifestParseError {
 
-        let manifest = CargoManifestParser.parse::<Document>(contents).unwrap();
-
-        let manifest = CargoManifest::try_from(manifest).unwrap();
+        let manifest = CargoManifest::try_from(metadata).unwrap();
         let version = manifest.minimum_rust_version.unwrap();
 
         assert_eq!(version, BareVersion::ThreeComponents(1, 56, 0));
     }
 
     #[test]
     fn parse_rust_version_three_components_with_pre_release() {
-        let contents = r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
-rust-version = "1.56.0-nightly"
-
-[dependencies]
-"#;
-
-        let manifest = CargoManifestParser.parse::<Document>(contents).unwrap();
-
-        let parse_err = CargoManifest::try_from(manifest).unwrap_err();
-
-        let ManifestParseError { inner } = parse_err;
-        assert_eq!(inner, Error::PreReleaseModifierNotAllowed);
+        let metadata = metadata_json(Some("1.56.0-nightly"), None);
+        // cargo_metadata will check this is invalid for us
+        let err = serde_json::from_str::<Metadata>(&metadata).unwrap_err();
+        assert_eq!(
+            err.to_string(),
+            "pre-release identifiers are not supported in rust-version at line 8 column 38"
+        );
     }
 
     #[test]
     fn parse_rust_version_two_components() {
-        let contents = r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
-rust-version = "1.56"
-
-[dependencies]
-"#;
-
-        let manifest = CargoManifestParser.parse::<Document>(contents).unwrap();
+        let metadata = metadata_json(Some("1.56"), None);
+        let metadata: Metadata = serde_json::from_str(&metadata).unwrap();
 
-        let manifest = CargoManifest::try_from(manifest).unwrap();
+        let manifest = CargoManifest::try_from(metadata).unwrap();
         let version = manifest.minimum_rust_version.unwrap();
 
-        assert_eq!(version, BareVersion::TwoComponents(1, 56));
-    }
-
-    #[yare::parameterized(
-        empty = {""},
-        one_component = {"1"},
-        one_component_dot = {"1."},
-        two_components_dot = {"1.1."},
-        three_components_dot = {"1.1.1."},
-        two_components_with_pre_release = {"1.1-nightly"},
-        two_components_not_a_number = {"1.x"},
-        three_components_not_a_number = {"1.1.x"},
-        too_many_components = {"1.1.0.0"},
-    )]
-    fn parse_rust_version_faulty_versions(version: &str) {
-        let contents = format!(
-            r#"[package]
-name = "some"
-version = "0.1.0"
-edition = "2018"
-rust-version = "{}"
-
-[dependencies]
-"#,
-            version
-        );
-
-        let manifest = CargoManifestParser.parse::<Document>(&contents).unwrap();
-
-        let manifest = CargoManifest::try_from(manifest);
-
-        assert!(manifest.is_err());
+        // going through cargo_metadata means it gets turned into 3 components
+        assert_eq!(version, BareVersion::ThreeComponents(1, 56, 0));
```

Comment:
This is also worth noting, not sure if it'll have any consequences?

Reply:
Should be fine as long as Rust sticks with semver :)